src/interoperability/interoperability.md

```diff
@@ -1,3 +1,27 @@
 # Interoperability
 
-> ❌: This section has not yet been written. Please refer to [embedded-wg#48](https://github.com/rust-lang-nursery/embedded-wg/issues/48) for discussion of this section.
\ No newline at end of file
+Iteroperability between Rust and C code is always dependant 
+on transforming data between the two languages.
+For this purposes there are two dedicated modules 
+in the `stdlib` called `std::ffi` and `std::raw`.
+
+For `#![no_std]` environments, there is also `core::raw` that
+covers all the primitive types.
+
+------------------------------------------
+| Rust type | Intermediate | C type       |
+|-----------|--------------|--------------|
+| String    | CString      | *char        |
+| &str      | CStr         | *const char  |
+| ()        | c_void       | void         |
+| u32 | u64 | c_uint       | unsigned int |
+| etc       | ...          | ...          |
+
+Primitive types can simply be cast to Rust types.
```

Comment:
Maybe instead of "cast" the wording "converted into" would be more appropriate for the example shown.  In this case, `c_uint` and `u32` end up being the same type but that won't always be the case.

Just thinking the audience might react negatively to the idea of a "cast" being required (which often implies unsafety).